### PR TITLE
Test dependency range and unbound install requirements

### PR DESCRIPTION
**Pull request by arcondello**
More-or-less just copies the `.circleci/config.yml` from https://github.com/dwavesystems/penaltymodel/pull/148 which was itself copied from the cloud-client I think?

Test oldest and latest versions of dependencies.
Test Python 3.11 and 3.12.
Test windows on circleci rather than appveyor.
Fix the doctests which apparently weren't running.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ Stack Overflow provides an excellent guide on [how to create a Minimal, Complete
 A clear and concise description of what you expected to happen.
 
 **Environment**
- - OS: [e.g., Ubuntu 16.04.4 LTS]
- - Python version: [e.g., 3.7.0]
+ - OS: [e.g. Ubuntu 22.04.3 LTS]
+ - Python version: [e.g. 3.12.0]
```
Comment (thisac) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]: 🎉

**Review by thisac — APPROVED** [review]
Looks good as far as I can tell.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,200 +1,259 @@
...
+          name: Run Python tests
+          command: env/bin/coverage run -m unittest discover
 
       - run:
-          name: linkcheck
+          name: Upload code coverage
           command: |
             . env/bin/activate
-            make -C docs/ linkcheck
+            codecov     # calls `coverage xml`, so we must activate venv
```
Comment (randomir) [inline .circleci/config.yml:58]: Maybe you want to use the new [codecov uploader](https://docs.codecov.com/docs/codecov-uploader), and their [orb](https://github.com/dwavesystems/dwave-cloud-client/blob/5829a538b2045c9fac1bec7fe7888feb6973945d/.circleci/config.yml#L75-L76). But it looks they backed down from the hard deprecation of old uploaders.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,200 +1,259 @@
 version: 2.1
+
+orbs:
+  win: circleci/windows@2.2.0
```
Comment (randomir) [inline .circleci/config.yml:4]:
```suggestion
  win: circleci/windows@5.0
```

**Review by randomir — APPROVED** [review]
lgtm

**Pull request merged**